src/InputReader.cpp

```diff
@@ -444,6 +444,46 @@ SparseDataset InputReader::read_movielens_ratings(const std::string& input_file,
   return SparseDataset(sparse_ds);
 }
 
+// TODO: Consolidate with read MovieLens function.
+SparseDataset InputReader::read_jester_ratings(const std::string& input_file,
+                                               int* number_users,
+                                               int* number_jokes) {
+  std::ifstream fin(input_file, std::ifstream::in);
+  if (!fin) {
+    throw std::runtime_error("Error opening input file " + input_file);
+  }
+
+  *number_jokes = *number_users = 0;
+
+  std::vector<std::vector<std::pair<int, FEATURE_TYPE>>> sparse_ds;
+
+  // XXX Fix
+  sparse_ds.resize(10000);
```

Comment:
I think we should create a variable in constants.h for this 10000.

Reply:
I think `Constants.h` might be the wrong scope given that the constant will only be used in `InputReader.cpp`. I'm also thinking about deleting the `Constants.h` file or otherwise moving other constants there, since after the FlatBuffers PR, the only constant in that file (the operations enum) will be removed. In the meantime I've added the 10000 constant to the top of InputReader.